plone/app/linkintegrity/tests/test_references.py

```diff
@@ -114,9 +114,15 @@ def test_broken_references(self):
                          [self.portal.doc1a])
 
         # Now delete the target item, suppress events and test again,
-        # the reference should be gone now.
+        # The reference should be a ghost not in any folder anymore.
+        # check if it has no acquition parent!
         self.portal._delObject(doc1a.id, suppress_events=True)
-        self.assertEqual([l.to_object for l in getOutgoingLinks(doc1)], [None])
+        obj = [l.to_object for l in getOutgoingLinks(doc1)]
+        if obj is not None:
```

Comment:
I expect this to fail, because `obj` is not `None`, but a list with either `[None]` or `[ghost]`. So we probably need to change this to: `for obj in [l.to_object for l in getOutgoingLinks(doc1)]:`.

Reply:
I guess the test might accidentally pass, but that is because the list has no `aq_parent`.

Reply:
agree, I'll fix this

Reply:
Perfect, thanks.